packages/bridge-ui/src/components/Transactions/Transaction.svelte

```diff
@@ -274,29 +283,35 @@
               controlsLayout={[]} />
           </div>
         {:else if transaction.receipt && [MessageStatus.New, MessageStatus.ClaimInProgress].includes(transaction.status)}
-          <button
-            class="cursor-pointer border rounded p-1 btn btn-sm border-white disabled:border-gray-800"
-            on:click={async () => await claim(transaction)}
+          <!-- 
+            TODO: we need some destructuring here. 
+                  We keep on accessing transaction props
+                  over and over again.
+          -->
+          <Button
+            type="accent"
+            size="sm"
+            on:click={onClaimClick}
             disabled={transaction.status === MessageStatus.ClaimInProgress}>
             Claim
-          </button>
```

Comment:
What is MessageStatus.ClaimInProgress?the MessageStatuses map to MessageStatuses from the contract, and this isn't one?

Reply:
Apparently this is for UI, but I'm not very sure. Maybe @shadab-taiko can answer that. I'll investigate how this is used though...

Reply:
@cyberhorsey The main idea was to disable the Claim button if the user has already clicked on it once. This wasn't meant to be the final solution for this since this would change once we start tracking claim transactions.